posthog/hogql/database/database.py

```diff
@@ -370,6 +370,52 @@ def _use_error_tracking_issue_id_from_error_tracking_issue_overrides(database: D
     )
 
 
+def _setup_group_key_fields(database: Database, team: "Team") -> None:
+    """
+    Set up group key fields as ExpressionFields that handle filtering based on GroupTypeMapping.created_at.
+    For $group_N fields, this returns:
+    - Empty string if no GroupTypeMapping exists for that index
+    - if(timestamp < mapping.created_at, '', $group_N) if GroupTypeMapping exists
+    """
+    group_mappings = {mapping.group_type_index: mapping for mapping in GroupTypeMapping.objects.filter(team=team)}
+
+    for group_index in range(5):
+        field_name = f"$group_{group_index}"
+
+        # If no mapping exists, leave the original field unchanged
+        if group_index in group_mappings:
+            group_mapping = group_mappings[group_index]
+
+            if group_mapping.created_at is None:
+                # If no created_at, just use the regular field - no need to replace
+                continue
+            else:
```

Comment:
nit: this else is not required, considering we're `continue`'ing above - just means we can de-indent this block a little